internal/sauceignore/matcher.go

```diff
@@ -0,0 +1,78 @@
+package sauceignore
+
+import (
+	"bufio"
+	"os"
+	"path/filepath"
+	"strings"
+
+	"github.com/go-git/go-git/v5/plumbing/format/gitignore"
+)
+
+const commentPrefix = "#"
+const sauceignore = ".sauceignore"
+
+// ReadIgnoreFile reads .sauceignore file and creates ignore patters if .sauceignore file is exists.
+func ReadIgnoreFile(path string) ([]Pattern, error) {
+	fPath := filepath.Join(path, sauceignore)
+	f, err := os.Open(fPath)
+	if err != nil {
+		// In case if .sauceignore file doesn't exists.
+		if os.IsNotExist(err) {
+			return []Pattern{}, nil
+		}
+		return []Pattern{}, err
+	}
+	defer f.Close()
+
+	ps := []Pattern{}
+	scanner := bufio.NewScanner(f)
+	for scanner.Scan() {
+		s := scanner.Text()
+		if !strings.HasPrefix(s, commentPrefix) && len(strings.TrimSpace(s)) > 0 {
+			ps = append(ps, NewPattern(s, nil))
+		}
+	}
+
+	return ps, nil
+}
+
+// Pattern defines a single sauceignore pattern.
+type Pattern struct {
+	P      string
+	Domain []string
+}
+
+// NewPattern create new Pattern.
+func NewPattern(p string, domain []string) Pattern {
```

Comment:
What's `domain`? It's not documented nor used 🤔

Reply:
For our needs, we can skip this field. In general, it's a domain (parent path) to which belongs this pattern.